text/0517-io-os-reform.md

```diff
@@ -547,4 +767,53 @@ principles or visions) are outside the scope of this RFC.
 # Unresolved questions
 [Unresolved questions]: #unresolved-questions
 
-> To be expanded in a follow-up PR.
+> To be expanded in follow-up PRs.
+
+## Wide string representation
+
+(Text from @SimonSapin)
+
+Rather than WTF-8, `OsStr` and `OsStrBuf` on Windows could use
+potentially-ill-formed UTF-16 (a.k.a. "wide" strings), with a
+different cost trade off.
+
+Upside:
+* No conversion between `OsStr` / `OsStrBuf` and OS calls.
+
+Downsides:
+* More expensive conversions between `OsStr` / `OsStrBuf` and `str` / `String`.
+* These conversions have inconsistent performance characteristics between platforms. (Need to allocate on Windows, but not on Unix.)
```

Comment:
Personally I'd argue that the inconsistent performance characteristics exist either way. The difference is whether they exist on OS calls, or on `OsStrBuf` creation. It seems a to me that it is not unlikely that the same `OsStr` would be used to call a function multiple times, or to call multiple different functions. To me this strongly suggests putting the cost at creation time, not call time.